Source/Dafny/Compiler-cpp.cs

```diff
@@ -1573,7 +1586,19 @@ protected class ClassWriter : IClassWriter {
       if (e is StaticReceiverExpr) {
         wr.Write(TypeName(e.Type, wr, e.tok));
       } else if (e.Value == null) {
-        wr.Write("nullptr");
+        var xType = e.Type.NormalizeExpand();
```

Comment:
Can we turn this whole block into a call to `EmitNull` to avoid the code duplication?